python/lsst/analysis/tools/actions/plot/histPlot.py

```diff
@@ -0,0 +1,319 @@
+            handles, labels = ax.get_legend_handles_labels()
+            all_handles += handles
+            all_nums += nums
+            all_meds += meds
+            all_mads += mads
+
+        # add side panel; add statistics
+        self._addStatisticsPanel(side_fig, all_handles, all_nums, all_meds, all_mads)
+
+        # add general plot info
+        hist_fig = addPlotInfo(hist_fig, plotInfo)
+
+        # finish up
+        hist_fig.text(0.01, 0.42, "Frequency", rotation=90, transform=hist_fig.transFigure)
+        plt.draw()
+        return fig
+
+    def _makeAxes(self, fig):
+        """Determine axes layout for main histogram figure."""
+        num_panels = len(self.panels)
+        if num_panels <= 1:
+            ncols = 1
+        else:
+            ncols = 2
+        nrows = int(np.ceil(num_panels / ncols))
+
+        gs = GridSpec(nrows, ncols, left=0.13, right=0.99, bottom=0.1, top=0.88, wspace=0.25, hspace=0.45)
+
+        axs = []
+        counter = 0
+        for row in range(nrows):
+            for col in range(ncols):
+                counter += 1
+                if counter < num_panels:
+                    axs.append(fig.add_subplot(gs[row : row + 1, col : col + 1]))
+                else:
+                    axs.append(fig.add_subplot(gs[row : row + 1, col : np.min([col + 2, ncols + 1])]))
+                    break
+
+        return axs
+
+    def _assignColors(self):
+        """Assign colors to histograms using a given color map."""
+        custom_cmaps = dict(
+            # https://www.tableau.com/about/blog/2016/7/colors-upgrade-tableau-10-56782
+            newtab10=[
+                "#4e79a7",
+                "#f28e2b",
+                "#e15759",
+                "#76b7b2",
+                "#59a14f",
+                "#edc948",
+                "#b07aa1",
+                "#ff9da7",
+                "#9c755f",
+                "#bab0ac",
+            ],
+            # https://personal.sron.nl/~pault/#fig:scheme_bright
+            bright=[
+                "#4477AA",
+                "#EE6677",
+                "#228833",
+                "#CCBB44",
+                "#66CCEE",
+                "#AA3377",
+                "#BBBBBB",
+            ],
+            # https://personal.sron.nl/~pault/#fig:scheme_vibrant
+            vibrant=[
+                "#EE7733",
+                "#0077BB",
+                "#33BBEE",
+                "#EE3377",
+                "#CC3311",
+                "#009988",
+                "#BBBBBB",
+            ],
+        )
+        if self.cmap in custom_cmaps.keys():
+            all_cols = custom_cmaps[self.cmap]
+        else:
+            try:
+                all_cols = getattr(plt.cm, self.cmap).copy().colors
+            except AttributeError:
+                raise ValueError(f"Unrecognized color map: {self.cmap}")
+
+        counter = 0
+        cols = defaultdict(list)
+        for panel in self.panels:
```

Comment:
Personally, I would prefer to have the colors repeat in each panel, so that, for example, psfFlux is always one color and apFlux is another. However, I realize that the panels are not guaranteed to have analogous datatypes, so maybe this would be annoying to implement.

Reply:
Right - there's no guarantee that hist1 in panel1 will correspond to a matching dataset type to that for hist 1 in panel 2. Also, repeating colors for each panel was how `HistPlotTask` was originally set up back in `analysis_drp`, however, it was decided that plot statistics should be printed in their own table on the side of the image, rather than printed into the corner of each panel. As such, repeating colors per-panel would make it confusing associating statistics with any given histogram.

If however you think nestling plot statistics into the corner of each panel would work well, please do let me know. We could re-add this functionality rather trivially (perhaps even with a config setting), which would then allow once more for per-panel color schemes. This effort may have to be pushed back to a follow-up ticket however.

Reply:
No, I wasn't thinking about what repeating colors would mean for the plot statistics panel. It is very readable as-is, so I think it is best to keep it that way.